### DISCUSSION TITLE
Packages take a long time to become visible to conda search after upload

### DISCUSSION THREAD

**Discussion by duncanmmacleod** [conda/infra#13] [General]
I regularly see that packages that get uploaded to https://anaconda.org/conda-forge take ~1 hour or more (as timed by the _Uploaded_ timestamp on the `/files` page for that package) to become visible with `conda search`. Is there an way to more regularly update the package list (repodata?) for a channel to have packages become usable with a lower latency?

My current example is `osx-64/python-lalburst-1.5.9-py310h93fb10b_2.tar.bz2` which was _Uploaded_ '1 hour and 6 minutes ago' by `cf-staging`, but isn't available from `conda search`

```console
$ date && conda search python-lalburst=1.5.9=py310h93fb10b_2
Fri Dec 10 07:19:40 PST 2021
Loading channels: done
No match found for: python-lalburst==1.5.9=py310h93fb10b_2. Search: *python-lalburst*==1.5.9=py310h93fb10b_2

PackagesNotFoundError: The following packages are not available from current channels:

  - python-lalburst==1.5.9=py310h93fb10b_2

Current channels:

  - https://conda.anaconda.org/conda-forge/osx-64
  - https://conda.anaconda.org/conda-forge/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

Is there anything that I ('the user') can do to speed things up?

**Comment by duncanmmacleod** [comment]
Update: this package became available around about 0729 PST (December 10), so approximately 75 minutes after being uploaded.

**Comment by jakirkham** [comment]
Packages that are uploaded to `conda-forge` are mirrored to a CDN. As it takes some time to do the mirroring, they don't show up immediately.

That said, it is typically closer to ~30mins. 75mins does sound a bit long. @chenghlee do you have any thoughts here?

**Comment by chenghlee** [comment]
Digging through the logs on the sync worker, since 2021-09, each CDN sync run now takes ~47 minutes; about 60% of that time (~28 min) is spent running `conda index`.

These numbers seem to be mostly independent of the number of packages being synced, so there's definitely something a little to a lot weird going on. I've opened a ticket with Anaconda's Infrastructure team to take a look at what's going on, though given the time of year, we may not have a real answer until Q1 of next year.

**Reply by duncanmmacleod** [reply]
As far as I can recall it's been pretty slow for a while, so waiting O(weeks) for an update is fine, happy to hear there may be some potential for improvement.

**Reply by isuruf** [reply]
@chenghlee, any updates on this?

**Reply by IvoLeist** [reply]
Hello, I am just curious if it is still expected that it takes so long till packages can be downloaded from conda-forge?
My package in question is: https://anaconda.org/channels/conda-forge/packages/iso639-lang which got
onto conda-forge on Jun 12, 2026 at 08:36 PM. However now after more than 1h past it still cannot be downloaded...

`date && mamba repoquery search iso639-lang --channel conda-forge`
> Fri Jun 12 09:39:11 PM CEST 2026
> Getting repodata from channels...
> 
> warning  libmamba 'repo.anaconda.com', a commercial channel hosted by Anaconda.com, is used.
>     
> warning  libmamba Please make sure you understand Anaconda Terms of Services.
>     
> warning  libmamba See: https://legal.anaconda.com/policies/en/
> ⚠ Shard Index for pkgs/main/linux-64 not available, falling back to flat repodata
> Using Flat Repodata for pkgs/main/linux-64                                                ✔ Done (0.2 sec)
> ⚠ Shard Index for pkgs/main/noarch not available, falling back to flat repodata
> Using Flat Repodata for pkgs/main/noarch                                                  ✔ Done (0.0 sec)
> ⚠ Shard Index for pkgs/r/linux-64 not available, falling back to flat repodata
> Using Flat Repodata for pkgs/r/linux-64                                                   ✔ Done (0.0 sec)
> ⚠ Shard Index for pkgs/r/noarch not available, falling back to flat repodata
> Using Flat Repodata for pkgs/r/noarch                                                     ✔ Done (0.0 sec)
> Using Cached Shard Index for conda-forge/linux-64                                                   ✔ Done
> Using Cached Shard Index for conda-forge/noarch                                                     ✔ Done
> Fetching and Parsing Packages' Shards                                                     ✔ Done (0.0 sec)
> Using Flat Repodata for conda-forge/linux-64                                              ✔ Done (1.6 sec)
> Using Cached Shard Index for conda-forge/noarch                                                     ✔ Done
> Using Cached Shard Index for conda-forge/linux-64                                                   ✔ Done
> Fetching and Parsing Packages' Shards                                                     ✔ Done (0.0 sec)
> Using Flat Repodata for conda-forge/noarch                                                ✔ Done (0.9 sec)
> Parsing Packages' Records                                                                 ✔ Done (3.2 sec)
> No entries matching "iso639-lang" found

**Reply by atravitz** [reply]
I'm seeing similar behavior >2 hrs waiting for the latest build of https://anaconda.org/channels/conda-forge/packages/gufe, which showed up on conda-forge at 14:50 UTC, and as of 17:07 UTC the newest build (`pyhc364b38_1.conda`) is not available:

```
date -z utc && micromamba search gufe=1.12 --channel conda-forge
Tue Jun 30 17:07:32 UTC 2026
Getting repodata from channels...

conda-forge/noarch                                          Using cache
conda-forge/osx-arm64                                       Using cache
nodefaults/osx-arm64                               125.0 B @ 687.0 B/s  0.0s
nodefaults/noarch                                   ??.?MB @  ??.?MB/s  0.0s
 Name Version Build          Channel     Subdir
────────────────────────────────────────────────
 gufe 1.12.0  pyhc364b38_0   conda-forge noarch
```

It would be helpful to know how long I should wait until I should assume something is broken.

**Comment by beepboop2025** [comment]
Another current data point (2026-08-28):

- Package: conda-forge/liquilens-evidence 0.15.0, build pyhc364b38_0
- Feedstock build succeeded: https://github.com/conda-forge/liquilens-evidence-feedstock/actions/runs/33152963342
- Anaconda upload time: 07:51:46 UTC
- More than one hour later, the package API and the explicit conda-forge/label/main artifact return 200, and the artifact SHA-256 is 5995bb7250dc34dc94715825e82038a1a48441d68950ccfa2f6a7d708f0f7877.
- The ordinary conda-forge/noarch artifact URL still returns 404, while current_repodata.json and channeldata.json do not contain liquilens-evidence.
- A fresh micromamba solve using only -c conda-forge cannot resolve liquilens-evidence=0.15.0. The same public artifact installs successfully from the explicit label/main channel and both package entry points pass smoke tests.
- Anaconda status currently reports all systems operational.

Package page: https://anaconda.org/conda-forge/liquilens-evidence
Feedstock: https://github.com/conda-forge/liquilens-evidence-feedstock

Is a delay beyond one hour currently expected for default-channel replication/indexing, or is there a specific service or queue maintainers should inspect?
